src/Xrefcheck/Config.hs

```diff
@@ -62,8 +61,86 @@ data NetworkingConfig' f = NetworkingConfig
   , ncMaxRetries                :: Field f Int
     -- ^ How many attempts to retry an external link after getting
     -- a "429 Too Many Requests" response.
+  , ncExternalRefRedirects      :: RedirectConfig
+    -- ^  Rules to override the redirect behavior for external references.
+  } deriving stock (Generic)
+
+doesMatchAnyRegex :: Text -> ([Regex] -> Bool)
+doesMatchAnyRegex src = any $ \regex ->
+  case regexec regex src of
+    Right res -> case res of
+      Just (before, match, after, _) ->
+        null before && null after && not (null match)
+      Nothing -> False
+    Left _ -> False
+
+-- | A list of custom redirect rules.
+type RedirectConfig = [RedirectRule]
```

Comment:
Should I move all this new redirect-related code to a dedicated module, or it is fine to let it here?